skiko/src/commonMain/kotlin/org/jetbrains/skia/paragraph/TextStyle.kt

```diff
@@ -110,10 +110,24 @@ class TextStyle internal constructor(ptr: NativePointer) : Managed(ptr, _Finaliz
         }
     }
 
-    var decorationStyle: org.jetbrains.skia.paragraph.DecorationStyle
+    var decorationStyle: DecorationStyle
         get() = try {
             Stats.onNativeCall()
-            _nGetDecorationStyle(_ptr)
+            val decorationStyleData = withResult(IntArray(4)) {
+                _nGetDecorationStyle(_ptr, it)
+            }
+
+            val decorationStyleFlags = decorationStyleData[0]
+
+            DecorationStyle(
```

Comment:
perhaps making some static method in `DecorationStyle.kt` to create `DecorationStyle` from  decorationStyleData might help to reuse this logic in other places when/if needed?